unittests/block_header_tests.cpp

```diff
@@ -0,0 +1,112 @@
+#include <eosio/chain/block_header.hpp>
+
+#include <boost/test/unit_test.hpp>
+
+using namespace eosio::chain;
+
+BOOST_AUTO_TEST_SUITE(block_header_tests)
+
+// test for block header without extension
+BOOST_AUTO_TEST_CASE(block_header_without_extension_test)
+{
+   block_header header;
+   std::optional<block_header_extension> ext = header.extract_header_extension(instant_finality_extension::extension_id());
+   BOOST_REQUIRE(!ext);
+}
+
+// test for empty instant_finality_extension
+BOOST_AUTO_TEST_CASE(instant_finality_extension_with_empty_values_test)
+{
+   block_header header;
+   constexpr uint32_t                    last_qc_block_num {0};
+   constexpr bool                        is_last_qc_strong {false};
+   const std::optional<finalizer_policy> new_finalizer_policy {std::nullopt};
+   const std::optional<proposer_policy>  new_proposer_policy {std::nullopt};
+
```

Comment:
same unneeded `{std::nullopt}` initializers. But even why define variables for these?

Reply:
For easier to see parameters when used in `pack`.

Reply:
Honestly, I'd prefer this:
`fc::raw::pack( instant_finality_extension{0 /*last_qc_block_num*/, false/*is_last_qc_strong*/, finalizer_policy{}, proposer_policy{}}`

Reply:
Different style. `/*` is not widely used in the code space.

Reply:
I removed `new_finalizer_policy` and `new_proposer_policy` but kept `last_qc_block_num` and `is_last_qc_strong`, as they are used later for comparison.